mlir/lib/Dialect/Mesh/Transforms/ShardingPropagation.cpp

```diff
@@ -34,6 +34,48 @@ namespace {
 // Utilities
 //===----------------------------------------------------------------------===//
 
+// This method returns all possible sharding attributes. For example,
+// mustShardings = [shard0, None] and optionalShardings = [None, shard1], the
+// result will be [[shard0, shard1], [shard0, None]]
+static SmallVector<SmallVector<MeshShardingAttr>>
+getOrderedPossibleShardingAttrs(ArrayRef<MeshShardingAttr> mustShardings,
```

Comment:
Could you mention in the comments what is the ordering?

Reply:
Done.